### PR TITLE
minecraft namespace

### PR DESCRIPTION
**Pull request by durandom**
https://github.com/operate-first/support/issues/139

create the namespace and groups on zero cluster

cc @goern

### CODE REVIEW

**Comment by durandom** [comment]
still requires the users to be added to the group

File: cluster-scope/overlays/moc/common/groups/b4mad.enc.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+apiVersion: user.openshift.io/v1
+kind: Group
+metadata:
+    name: b4mad
```
Comment (durandom) [inline cluster-scope/overlays/moc/common/groups/b4mad.enc.yaml:4]: @tumido not sure if the following would be needed here 

```
    annotations:
        kustomize.config.k8s.io/behavior: replace
```

**Comment by durandom** [comment]
@harshad16 thanks, I've included your commit from https://github.com/operate-first/apps/pull/416 here, so you can close your PR.

/unhold

@tumido @HumairAK please review

File: cluster-scope/overlays/moc/common/groups/b4mad.enc.yaml (hunk 1 above)
Comment (tumido) [reply cluster-scope/overlays/moc/common/groups/b4mad.enc.yaml:4]: It does to be there, that's a directive for `ksops` to merge on the group from `base`:

```
$ kustomize build --enable_alpha_plugins cluster-scope/overlays/moc/zero
2021/03/21 13:02:26 Attempting plugin load from '/home/tcoufal/.config/kustomize/plugin/viaduct.ai/v1/ksops/ksops.so'
Error: accumulating resources: 2 errors occurred:
        * accumulateFile error: "accumulating resources from '../common': '/home/tcoufal/Programming/AI-CoE/operate-first/apps/cluster-scope/overlays/moc/common' must resolve to a file"
        * accumulateDirector error: "recursed accumulation of path '/home/tcoufal/Programming/AI-CoE/operate-first/apps/cluster-scope/overlays/moc/common': merging from generator &{0xc0003904e0 0xc0006482a0 [groups/ai-services.enc.yaml groups/argocd-admins.enc.yaml groups/b4mad.enc.yaml groups/data-science.enc.yaml groups/mesh-for-data.enc.yaml groups/open-aiops.enc.yaml groups/operate-first.enc.yaml groups/ray.enc.yaml groups/rekor.enc.yaml groups/thoth.enc.yaml groups/tufts-dcc-6.enc.yaml groups/workshops.enc.yaml]}: id resid.ResId{Gvk:resid.Gvk{Group:\"user.openshift.io\", Version:\"v1\", Kind:\"Group\"}, Name:\"b4mad\", Namespace:\"\"} exists; behavior must be merge or replace"
```

Once this annotation is added, the build succeeds.

**Review by tumido — CHANGES_REQUESTED** [review]

File: cluster-scope/overlays/moc/common/groups/b4mad.enc.yaml (hunk 1 above)
Comment (tumido) [reply cluster-scope/overlays/moc/common/groups/b4mad.enc.yaml:4]: @durandom you can't directly edit the encrypted manifest. Sops recognizes it was directly tampered with and refuses to decrypt it:

```
kustomize build --enable_alpha_plugins cluster-scope/overlays/moc/zero
2021/03/21 13:15:04 Attempting plugin load from '/home/tcoufal/.config/kustomize/plugin/viaduct.ai/v1/ksops/ksops.so'
Error: accumulating resources: 2 errors occurred:
        * accumulateFile error: "accumulating resources from '../common': '/home/tcoufal/Programming/AI-CoE/operate-first/apps/cluster-scope/overlays/moc/common' must resolve to a file"
        * accumulateDirector error: "recursed accumulation of path '/home/tcoufal/Programming/AI-CoE/operate-first/apps/cluster-scope/overlays/moc/common': trouble converting file 'groups/b4mad.enc.yaml' to a resource: trouble decrypting file groups/b4mad.enc.yaml: Failed to verify data integrity. expected mac \"E5E5FC869DF9AC50056830D1D41E4D127BDDFC2BC4DD8F9AE7CC5F2D2D65048CC012047C0305441DD21E164BECB6290AC72951BE3D33D8AAC987EB23FED4550C\", got \"025015EBA68FA023A54C70768FC326E4F1A3CC35BC62B62BF50EE84D51431117150FFA350438583136FB84B22A1B7C04406A882942FDB5773FFA3B773D3DAD13\""
```

I know the traceback is not very readable, the important part is:
```
Failed to verify data integrity. expected mac \"E5E5FC869DF9AC50056830D1D41E4D127BDDFC2BC4DD8F9AE7CC5F2D2D65048CC012047C0305441DD21E164BECB6290AC72951BE3D33D8AAC987EB23FED4550C\", got \"025015EBA68FA023A54C70768FC326E4F1A3CC35BC62B62BF50EE84D51431117150FFA350438583136FB84B22A1B7C04406A882942FDB5773FFA3B773D3DAD13\""
```

You have to edit the file via sops. Simply run following command against the previous revision of this file (without the annotation):

```
sops cluster-scope/overlays/moc/common/groups/b4mad.enc.yaml
```

**Review by tumido — CHANGES_REQUESTED** [review]

File: cluster-scope/overlays/moc/common/groups/b4mad.enc.yaml (hunk 1 above)
Comment (durandom) [reply cluster-scope/overlays/moc/common/groups/b4mad.enc.yaml:4]: 🤦

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Pull request merged**